store/postgres/src/relational/index.rs

```diff
@@ -517,13 +563,137 @@ impl CreateIndex {
             ) => {
                 unique == o_unique
                     && method == o_method
-                    && columns == o_columns
+                    && Expr::is_same_kind_columns(columns, o_columns)
                     && cond == o_cond
                     && with == o_with
             }
         }
     }
 
+    pub fn is_id_immutable_table(&self, immutable: bool) -> bool {
```

Comment:
Passing `immutable` in here feels a little strange. I think it would be better to just have this as `is_id(&self) -> bool` and have callers do `immutable && ci.is_id()`

Reply:
The motivation was to make the logic conditions simpler with helper, but I agree it looks silly. Changed it.